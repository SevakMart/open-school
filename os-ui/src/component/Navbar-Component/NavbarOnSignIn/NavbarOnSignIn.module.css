.mainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 28px 88px;
  box-sizing: border-box;
  background-color: #F9F9FD;
}
.mainContent h2 {
  font-family: "Helvetica Neue";
  font-style: italic;
  font-weight: 700;
  font-size: 20px;
  line-height: 24px;
  color: #222427;
  margin: 0;
  cursor: pointer;
}
.mainContent .navMainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.mainContent .navMainContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B;
  cursor: pointer;
  box-sizing: border-box;
  margin: 0 0 0 32px;
}
.mainContent .navMainContent .bookmarkIconStyle {
  margin: 0 0 0 25px;
  width: 20px;
  height: 20px;
  text-align: center;
}
.mainContent .navMainContent .notificationIcon {
  margin: 0 0 0 25px;
  cursor: pointer;
}
.mainContent .navMainContent .userInfoContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  margin: 0 0 0 25px;
}
.mainContent .navMainContent .userInfoContent .avatarLogo {
  width: 32px;
  height: 32px;
  border-radius: 50%;
}
.mainContent .navMainContent .userInfoContent .arrowDownIcon {
  cursor: pointer;
}/*# sourceMappingURL=NavbarOnSignIn.module.css.map */