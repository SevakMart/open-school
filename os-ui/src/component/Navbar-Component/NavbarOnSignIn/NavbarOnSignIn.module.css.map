{"version":3,"sources":["NavbarOnSignIn.module.scss","../../../styles/Mixins/FlexMixins.scss","NavbarOnSignIn.module.css","../../../styles/Mixins/FontMixin.scss","../../../styles/Configs/Colors.scss","../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAKA;ECJE,aAAA;EACA,mBDIqB;ECHrB,mBDG0B;ECF1B,8BDEkC;EAChC,kBAAA;EACA,sBAAA;EACA,yBAAA;AEDJ;AFEI;EGTF,6BAAA;EACA,kBHSyC;EGRzC,gBHQgD;EGPhD,eHOoD;EGNpD,iBHMyD;EACnD,cITY;EJUZ,SAAA;EACA,eAAA;AEIR;AFFI;ECfF,aAAA;EACA,mBDeyB;ECdzB,mBDc8B;ECb9B,8BDasC;AEOxC;AFNQ;EGjBN,yBEDY;EFEZ,kBHiB2C;EGhB3C,gBHgBkD;EGflD,eHesD;EGdtD,iBHc2D;EACjD,cIlBM;EJmBN,eAAA;EACA,sBAAA;EACA,kBAAA;AEYZ;AFVQ;EACI,kBAAA;EACA,WAAA;EACA,YAAA;EACA,kBAAA;AEYZ;AFVQ;EACI,kBAAA;EACA,eAAA;AEYZ;AFVQ;EClCN,aAAA;EACA,mBDkC6B;ECjC7B,mBDiCkC;EChClC,8BDgC0C;EAChC,sBAAA;EACA,kBAAA;AEeZ;AFdY;EACI,WAAA;EACA,YAAA;EACA,kBAAA;AEgBhB;AFdY;EACI,eAAA;AEgBhB","file":"NavbarOnSignIn.module.css"}