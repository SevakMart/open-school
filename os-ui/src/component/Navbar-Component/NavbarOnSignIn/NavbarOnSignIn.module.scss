@import "../../../styles/Mixins/ButtonMixin.scss";
@import "../../../styles/Mixins/FlexMixins.scss";
@import "../../../styles/Mixins/FontMixin.scss";
@import "../../../styles/Configs/Colors.scss";
@import "../../../styles/Configs/Fonts.scss";
.mainContent {
    @include flexStyle(row, center, space-between);
    margin:0 88px;
    padding:28px 0;
    background-color: transparent;
    h2 {
        @include fontStyle($secondary-font,italic,700,20px,24px);
        color:$header-main-color;
        margin:0;
    }
    .navMainContent{
        @include flexStyle(row, center, space-between);
        p{
            @include fontStyle($primary-font,normal,500,16px,24px);
            color:$main-text-color;
            cursor:pointer;
            box-sizing: border-box;
            margin:0 0 0 32px;
        }
        .bookmarkIconStyle {
            margin:0 0 0 25px;
            width:20px;
            height:20px;
            text-align: center;
        }
        .notificationIcon {
            margin:0 0 0 25px;
            cursor:pointer;
        }
        .userInfoContent {
            @include flexStyle(row, center, space-between);
            box-sizing: border-box;
            margin:0 0 0 25px;
            .avatarLogo {
                width:32px;
                height:32px;
                border-radius: 50%;
            }
            .arrowDownIcon {
                cursor:pointer;
            }
        }
    }
}
