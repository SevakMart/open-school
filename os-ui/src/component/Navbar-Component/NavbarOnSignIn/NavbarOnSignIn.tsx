import { useState, useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import styles from './NavbarOnSignIn.module.scss';
import ArrowDownIcon from '../../../assets/svg/Arrow-down.svg';
import NotificationIcon from '../../../assets/svg/Notification.svg';
import BookmarkIcon from '../../../icons/Bookmark';
import { SignoutIcon } from '../../../icons/Signout/Signout';
import ProfilePortalContent from '../../ProfilePortalContent/ProfilePortalContent';
import { Portal } from '../../Portal/Portal';

const NavbarOnSignIn = () => {
  const [isOpen, setIsOpen] = useState(false);
  const navigate = useNavigate();
  const arrowRef = useRef(null);
  const { t } = useTranslation();

  const {
    mainContent, navMainContent, userInfoContent,
    bookmarkIconStyle, notificationIcon, avatarLogo, arrowDownIcon,
  } = styles;

  const handleProfilePortal = () => {
    setIsOpen((prevState) => !prevState);
  };

  const popupRef = useRef<HTMLDivElement | null>(null);
  const bodyRef = useRef(document.body);

  const handleClickOutside = (event: MouseEvent) => {
    const target = event.target as Node;
    if (popupRef.current && !popupRef.current.contains(target)) {
      setIsOpen(false);
    }
  };

  useEffect(() => {
    const listener = (event: MouseEvent) => handleClickOutside(event);
    bodyRef.current.addEventListener('mousedown', listener);
    return () => {
      bodyRef.current.removeEventListener('mousedown', listener);
    };
  }, []);

  return (
    <nav className={mainContent}>
      <h2 onClick={() => navigate('/homepage/WhenSignin')}>{t('string.homePage.navBar.logo')}</h2>
      <div className={navMainContent}>
        <p onClick={() => navigate('/exploreLearningPaths')}>{t('string.homePage.navBar.allLearningPaths')}</p>
        <p onClick={() => navigate('/myLearningPath')}>{t('string.homePage.navBar.myLearningPaths')}</p>
        <p onClick={() => navigate('/mentors')}>{t('string.homePage.navBar.mentors')}</p>
        <div className={bookmarkIconStyle}>
          <BookmarkIcon iconSize="20px" />
        </div>
        <img className={notificationIcon} src={NotificationIcon} alt="Notification" />
        <div className={userInfoContent}>
          <img className={avatarLogo} src="https://reactjs.org/logo-og.png" alt="avatar" />
          <img ref={arrowRef} className={arrowDownIcon} style={isOpen ? { transform: 'rotate(180deg)' } : undefined} src={ArrowDownIcon} alt="Arrow down Icon" onClick={handleProfilePortal} />
        </div>
        <div ref={popupRef}>
          <Portal.ProfilePortal isOpen={isOpen}>
            <div ref={popupRef}>
              <ProfilePortalContent icon={<SignoutIcon />} isSignOut>
                {t('string.profilePortal.signOut')}
              </ProfilePortalContent>
            </div>
          </Portal.ProfilePortal>
        </div>
      </div>
    </nav>
  );
};
export default NavbarOnSignIn;
