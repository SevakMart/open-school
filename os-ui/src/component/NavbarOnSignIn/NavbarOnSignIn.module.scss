@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";
.mainContent {
    @include flexStyle(row, center, space-between);
    padding-left: 6%;
    padding-right: 6%;
    background-color: transparent;
    h2 {
        @include fontStyle('Helvetica Neue',italic,700,1.25rem,1.5rem);
        color:#222427;
    }
    .navMainContent{
        @include flexStyle(row, center, space-between);
        width:40%;
        p{
            @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
            color:#5E617B;
            cursor:pointer;
            margin-left: 1px;
        }
        .userInfoContent {
            @include flexStyle(row, center, space-between);
            width:10%;
            img {
                width:2rem;
                height:2rem;
                border-radius: 50%;
            }
        }
    }
}
