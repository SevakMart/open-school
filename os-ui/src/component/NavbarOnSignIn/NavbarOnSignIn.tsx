import styles from './NavbarOnSignIn.module.scss';
import {
  APP_LOGO, ALL_LEARNING_PATHS, MY_LEARNING_PATHS, MENTORS,
} from '../../constants/Strings';
import BookmarkIcon from '../../icons/Bookmark';
import Notification from '../../icons/Notification';
import DownArrowIcon from '../../icons/DownArrow';

const NavbarOnSignIn = () => {
  const { mainContent, navMainContent, userInfoContent } = styles;
  return (
    <nav className={mainContent}>
      <h2>{APP_LOGO}</h2>
      <div className={navMainContent}>
        <p>{ALL_LEARNING_PATHS}</p>
        <p>{MY_LEARNING_PATHS}</p>
        <p>{MENTORS}</p>
        <BookmarkIcon iconSize="1.5rem" isBookmarked={false} />
        <Notification />
        <div className={userInfoContent}>
          <img src="https://reactjs.org/logo-og.png" alt="avatar" />
          <DownArrowIcon />
        </div>
      </div>
    </nav>
  );
};
export default NavbarOnSignIn;
