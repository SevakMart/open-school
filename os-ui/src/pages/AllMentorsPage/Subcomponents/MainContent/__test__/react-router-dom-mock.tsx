export const mockUseNavigate = jest.fn();
export const reactRouterDomMock = {
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavigate,
  useLocation: () => ({ search: '' }),
};
