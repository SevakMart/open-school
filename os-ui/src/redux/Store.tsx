import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
