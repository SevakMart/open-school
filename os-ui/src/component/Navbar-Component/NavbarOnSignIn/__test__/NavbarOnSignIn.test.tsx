import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../../../redux/Store';
import NavbarOnSignIn from '../NavbarOnSignIn';
import { reactRouterDomMock } from '../../../../pages/AllMentorsPage/Subcomponents/MainContent/__test__/react-router-dom-mock';

jest.doMock('react-router-dom', () => reactRouterDomMock);
jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Make unit tests of NavbarOnSignIn component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <NavbarOnSignIn />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
