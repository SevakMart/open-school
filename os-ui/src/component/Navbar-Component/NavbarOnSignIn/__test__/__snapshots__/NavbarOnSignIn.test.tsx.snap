// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make unit tests of NavbarOnSignIn component Create a snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="navMainContent"
    >
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.allLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.myLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.mentors
      </p>
      <div
        class="bookmarkIconStyle"
      >
        <img
          alt="bookmark"
          src="bookMark.svg"
          style="font-size: 20px; cursor: pointer; fill: none;"
        />
      </div>
      <img
        alt="Notification"
        class="notificationIcon"
        src="Notification.svg"
      />
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          class="avatarLogo"
          src="https://reactjs.org/logo-og.png"
        />
        <img
          alt="Arrow down Icon"
          class="arrowDownIcon"
          src="Arrow-down.svg"
        />
      </div>
      <div />
    </div>
  </nav>
</DocumentFragment>
`;
